Sources/Nimble/Matchers/ThrowAssertion.swift

```diff
@@ -27,21 +25,19 @@ public func throwAssertion() -> Predicate<Void> {
             #endif
             do {
                 try actualExpression.evaluate()
-            } catch let error {
-                succeeded = false
-                failureMessage.postfixMessage += "; threw error instead <\(error)>"
+            } catch {
+                actualError = error
             }
         }
 
-        if !succeeded {
-            return false
-        }
-
-        if caughtException == nil {
-            return false
+        if let actualError = actualError {
+            return PredicateResult(
+                bool: false,
+                message: message.appended(message: "; threw error instead <\(actualError)>")
+            )
         }
 
-        return true
+        return PredicateResult(bool: caughtException != nil, message: message)
```

Comment:
How would you feel about moving this return statement into an `else` branch above?